src/Native/Runtime/ThunksMapping.cpp

```diff
@@ -102,7 +104,9 @@ EXTERN_C REDHAWK_API void* __cdecl RhAllocateThunksMapping()
 
     // Note: On secure linux systems, we can't add execute permissions to a mapped virtual memory if it was not created 
     // with execute permissions in the first place. This is why we create the virtual section with RX permissions, then
-    // reduce it to RW for the data section and RX for the stubs section after generating the stubs instructions.
+    // reduce it to RW for the data section. For the stubs section we need to increase to RWX to generate the stubs
```

Comment:
RWX seems to be used only as a fallback when the OS does not allow RX -> RW transition. It came in the original implementation https://github.com/dotnet/corert/pull/2213/files#diff-37d78e01ff715e94386336522e610e7fR119 and there are no comments about, so it is hard to tell what it is needed for.

Ideally, we would never create RWX pages, some OSes even disallow it.

Does Android disallow RX -> RW transition?

Reply:
I didn't modify the code that changes the protection . I just updated the comment as it telling another story before and that confused me when I started to work on this. On Android it works fine going to RW and than back to RX. But as I said the RWX was already there so I kept it.

Reply:
The comment says "For the stubs section we need to increase to RWX" now that is also confusing. It should say that RWX is used as fallback when the OS does not support RX -> RW transition.

Reply:
I understand it the other way around. On a secure linux you cannot add execute rights later on. Therefore the section is already created as RX. The data section is than downgraded to RW. But to fill in the thunks code we need to get write privileges for this section. So it's increased to RWX. After the code is written is goes back to RX. I assumed that on a secure Linux you can not add the X back after you have dropped it and therefore who ever has written this code decided to go RX->RWX->RX for the stubs. 

I am fine with puting the old comment back that didn't mentioned the RWX as I didn't change the part of the code it belongs, too 

Reply:
Ah, this whole paragraph talks about secure Linux. It looks good then.